File: mobilne/lab1/zad1Java/app/src/main/java/mobilki/zad1java/MainActivity.java
package mobilki.zad1java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    int random_left;
    int random_right;
    int points;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setListeners();
        roll();
    }


    protected void roll(){
        Button left_button = (Button) findViewById(R.id.leftButton);
        Button right_button = (Button) findViewById(R.id.rightButton);
        Random rng = new Random();
        random_left = rng.nextInt(10);
        random_right = rng.nextInt(10);
        left_button.setText(String.valueOf(random_left));
        right_button.setText(String.valueOf(random_right));
    }

    protected void checkGuess(int answer, int other ){
        if(answer > other){
            Toast.makeText(this, "Correct!",Toast.LENGTH_SHORT).show();
            points++;
            TextView pointText = findViewById(R.id.Points);
            pointText.setText(getString(R.string.gamePoints, points));
        }else{
            Toast.makeText(this, "Wrong!",Toast.LENGTH_SHORT).show();
        }
        roll();
    }
    protected void setListeners(){
        Button left_button = findViewById(R.id.leftButton);
        Button right_button = findViewById(R.id.rightButton);

        findViewById(R.id.leftButton).setOnClickListener(v -> checkGuess(random_left,random_right));
        findViewById(R.id.rightButton).setOnClickListener(v -> checkGuess(random_right,random_left));

    }


}

File: mobilne/lab1/zad1/app/src/main/java/lab1/zad1/MainActivity.kt
package lab1.zad1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    private var rollMax = 10
    private var points = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setClickListeners()
        generateNumbers()
        findViewById<TextView>(R.id.Points).text = getString(R.string.points_string,0)
    }

    private fun generateNumbers() {
        val r1 = Random.nextInt(rollMax)
        val r2 = Random.nextInt(rollMax)
        findViewById<Button>(R.id.buttonLeft).text = r1.toString()
        findViewById<Button>(R.id.buttonRight).text = r2.toString()
    }


    private fun checkAnswer(answer: Int, other: Int) {
        if (answer > other) {
            Toast.makeText(this@MainActivity, "Correct!", Toast.LENGTH_SHORT).show()
            points++
            findViewById<TextView>(R.id.Points).text = getString(R.string.points_string, points)
        } else {
            Toast.makeText(this@MainActivity, "Wrong!", Toast.LENGTH_SHORT).show()
        }
        generateNumbers()
    }

    private fun setClickListeners() {
        val leftButton = findViewById<Button>(R.id.buttonLeft)
        val rightButton = findViewById<Button>(R.id.buttonRight)
        leftButton.setOnClickListener {
            checkAnswer(
                Integer.parseInt(leftButton.text.toString()),
                Integer.parseInt(rightButton.text.toString())
            )
        }
        rightButton.setOnClickListener {
            checkAnswer(
                Integer.parseInt(rightButton.text.toString()),
                Integer.parseInt(leftButton.text.toString())
            )
        }
    }


}